Workbooks/MySQL Flexible Server/Auditing/Auditing.workbook

```diff
@@ -0,0 +1,225 @@
+      "name": "text - 4",
+      "styleSettings": {
+        "showBorder": true
+      }
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "AzureDiagnostics\r\n| where Category == 'MySqlAuditLogs'\r\n| project TimeGenerated, Resource, event_class_s, event_subclass_s, event_time_t, user_s , ip_s , sql_text_s \r\n| summarize count() by event_class_s, event_subclass_s\r\n| order by event_class_s",
+        "size": 1,
+        "timeContext": {
+          "durationMs": 86400000
+        },
+        "queryType": 0,
+        "resourceType": "microsoft.dbformysql/flexibleservers",
+        "visualization": "piechart",
+        "gridSettings": {
+          "sortBy": [
+            {
+              "itemKey": "event_class_s",
+              "sortOrder": 1
+            }
+          ]
+        },
+        "sortBy": [
+          {
+            "itemKey": "event_class_s",
+            "sortOrder": 1
+          }
+        ]
+      },
+      "customWidth": "50",
+      "name": "query - 6",
+      "styleSettings": {
+        "showBorder": true
+      }
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "AzureDiagnostics\r\n| where ResourceProvider == \"MICROSOFT.DBFORMYSQL\"\r\n| where Category == \"MySqlAuditLogs\" and event_class_s == \"connection_log\"\r\n| project TimeGenerated,event_subclass_s , ip_s, user_s , db_s ,connection_id_d\r\n| summarize count() by event_subclass_s ,bin(TimeGenerated, 5m)",
+        "size": 1,
+        "timeContext": {
+          "durationMs": 86400000
+        },
+        "queryType": 0,
+        "resourceType": "microsoft.dbformysql/flexibleservers",
+        "visualization": "piechart"
+      },
+      "customWidth": "50",
+      "name": "query - 9",
+      "styleSettings": {
+        "showBorder": true
+      }
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "**Audit Connection Events** (You should enable  MySQL Auditing  Ref: https://docs.microsoft.com/en-us/azure/mysql/flexible-server/concepts-audit-logs)",
+        "style": "info"
+      },
+      "customWidth": "50",
+      "name": "text - 4 - Copy",
+      "styleSettings": {
+        "showBorder": true
+      }
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "**Table Access Summary** (You should enable  MySQL Auditing  Ref: hhttps://docs.microsoft.com/en-us/azure/mysql/flexible-server/concepts-audit-logs)",
+        "style": "info"
+      },
+      "customWidth": "50",
+      "name": "text - 7",
+      "styleSettings": {
+        "showBorder": true
+      }
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "AzureDiagnostics\r\n| where ResourceProvider == \"MICROSOFT.DBFORMYSQL\"\r\n| where Category == \"MySqlAuditLogs\" and event_class_s == \"connection_log\"\r\n| project TimeGenerated,event_subclass_s , ip_s, user_s , db_s ,connection_id_d",
```

Comment:
unless you *directly* control the data going into these tables, you might want to use case-insensitive for text columns, with `=~` instead of `==` .  this is especially true if you do any testing against resource ids directly, in general azure resources are case insensitive but case preserving, and we've seen LOTS of cases where logs has many rows with inconsistent casing of resource ids.

you could also use `columnifexists` to protect yourself for missing schema
```
AzureDiagnostics 
|  where ResourceProvider =~ "MICROSOFT.DBFORMYSQL"
| where Category =~ "MySqlAuditLogs" and columnifexists("event_class_s", "") =~ "connection_log"
```

Reply:
We fixed all the queries with the suggestions.